core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultProgrammaticDriverConfigLoaderBuilder.java

```diff
@@ -46,6 +48,25 @@
 
   private String currentProfileName = DriverExecutionProfile.DEFAULT_NAME;
 
+  /**
+   * Builds a supplier that uses the default driver fallback config, but uses the supplied
+   * ClassLoader (as opposed to the Driver's ClassLoader) for application-specific config overrides.
+   *
+   * <p>This method is primarily intended for cases where programmatic config is desired, but you
+   * also want to provide application defaults that override Driver defaults, and the application is
+   * loaded by a different ClassLoader than the current context of the Driver's ClassLoader. This
+   * can be the situation in an OSGi application bundle that uses programmatic configuration with an
+   * {@code application.conf}.
+   *
+   * @param appClassLoader The application's ClassLoader from which to load application
+   *     configuration defaults.
+   */
+  public static Supplier<Config> getDefaultConfigSupplier(ClassLoader appClassLoader) {
+    return () ->
+        ConfigFactory.defaultApplication(appClassLoader)
+            .withFallback(ConfigFactory.defaultReference(DriverConfigLoader.DRIVER_CLASS_LOADER));
+  }
```

Comment:
This will behave identically to the `DEFAULT_FALLBACK_SUPPLIER` above, except that it will not rely on the current Context ClassLoader to find application overrides (i.e. `application.conf`), and will instead use the supplied ClassLoader for that.

Reply:
I don't think we should expose such a method, I would instead inline this code in the constructor below.

Reply:
I wasn't sure if it was a good idea to expose this. I will do as you suggest.